### PR TITLE
Add exec crash checks and minor cleanup

### PR DESCRIPTION
**Pull request by unknownbrackets**
When debugging, the crash checks were fairly annoying as the blue screen wouldn't go away, even on restart.  This fixes that.

It also tries to catch exec jump crashes, mainly on x86.  Didn't really do arm yet.

An easy way to test on arm/arm64:
 * Enable remote debugging on your phone.
 * Start a game and hit Break in the web debugger.
 * Replace the next instruction with `jr v0` or `j 0` or similar.

An edge case that is trickier (should probably setup a prx for it) is a *branch* into the wilderness.  This is possibly even more unlikely than a bad jump, but could happen if code was loaded into the scratchpad and branched from there, for example.

I didn't really test that, but I setup code that should catch it on x86 at least.

-[Unknown]

### CODE REVIEW

File: Core/HLE/sceKernelThread.cpp [hunk 1]
```diff
@@ -1462,6 +1468,12 @@ void __KernelLoadContext(PSPThreadContext *ctx, bool vfpuEnabled) {
 		memcpy(currentMIPS->vfpuCtrl, ctx->vfpuCtrl, sizeof(ctx->vfpuCtrl));
 	}
 
+#ifdef _DEBUG
```
Comment (hrydgard) [inline Core/HLE/sceKernelThread.cpp:1471]: This check is cheap enough that we might as well keep it in release too?

**Comment by hrydgard** [comment]
Some minor build errors, otherwise looking good!

**Comment by unknownbrackets** [comment]
ff1a31694 has the fix but GitHub isn't showing it.  Already pushed to the branch, sigh.

-[Unknown]

File: Core/HLE/sceKernelThread.cpp (hunk 1 above)
Comment (unknownbrackets) [reply Core/HLE/sceKernelThread.cpp:1471]: Well, was just being paranoid, but sure I'll remove the check.

-[Unknown]

**Comment by unknownbrackets** [comment]
Okay, reworded the commit message and GitHub is back with the program.

-[Unknown]
